functional-test/src/test/groovy/org/rundeck/tests/functional/api/job/JobExecutionSpec.groovy

```diff
@@ -1112,6 +1120,775 @@ class JobExecutionSpec extends BaseContainer {
+        ).status == ExecutionStatus.SUCCEEDED.state
+
+        when: "We run the job with notification"
+        def jobRun = JobUtils.executeJob(notificationJobId, client)
+        assert jobRun.successful
+        Execution parsedExecutionsResponse = mapper.readValue(jobRun.body().string(), Execution.class)
+
+        then: "Will succeed"
+        JobUtils.waitForExecutionToBe(
+                ExecutionStatus.SUCCEEDED.state,
+                parsedExecutionsResponse.id as String,
+                mapper,
+                client,
+                WaitingTime.LOW.milliSeconds,
+                WaitingTime.EXCESSIVE.milliSeconds / 1000 as int
+        ).status == ExecutionStatus.SUCCEEDED.state
+
+        when: "We wait 10 seconds for netcat to deliver response"
+        Thread.sleep(WaitingTime.MODERATE.milliSeconds * 2)
+
+        // Then run the job that reads the output of request
+        def readJobRun = JobUtils.executeJob(readNcOutputJobId, client)
+        assert readJobRun.successful
+        Execution readJobRunResponse = mapper.readValue(readJobRun.body().string(), Execution.class)
+        def readJobSucceeded = JobUtils.waitForExecutionToBe(
+                ExecutionStatus.SUCCEEDED.state,
+                readJobRunResponse.id as String,
+                mapper,
+                client,
+                WaitingTime.LOW.milliSeconds,
+                WaitingTime.MODERATE.milliSeconds / 1000 as int
+        )
+        assert readJobSucceeded.status == ExecutionStatus.SUCCEEDED.state
+        def execOutputResponse = client.doGetAcceptAll("/execution/$readJobRunResponse.id/output")
+        ExecutionOutput execOutput = mapper.readValue(execOutputResponse.body().string(), ExecutionOutput.class)
+
+        def webhookData = [
+            "POST /test?id=$parsedExecutionsResponse.id&status=succeeded HTTP/1.1".toString(),
+            "X-RunDeck-Notification-Execution-Status: succeeded",
+            "X-RunDeck-Notification-Execution-ID: $parsedExecutionsResponse.id".toString(),
+            "X-RunDeck-Notification-Trigger: success",
+            "Content-Type: text/xml; charset=UTF-8",
+            "Host: localhost:9001".toString(),
+            "<notification trigger='success' status='succeeded' executionId='$parsedExecutionsResponse.id'>".toString(),
+        ]
+        def entries = execOutput.entries.stream().map {it.log}.collect(Collectors.toList())
+
+        then: "The output of the job must have basic info about the webhook"
+        execOutput != null
+        FileHelpers.assertLinesInsideEntries(webhookData, entries)
+
+        cleanup:
+        deleteProject(projectName)
+        Files.deleteIfExists(Paths.get("/tmp/netcat-out.txt"))
+    }
+
+    @ExcludePro
+    def "test-job-run-without-deadlock"(){
+        setup:
+        def projectName = PROJECT_NAME
+        def client = getClient()
+        ObjectMapper mapper = new ObjectMapper()
+
+        def xmlJob = (String stepArgs) -> {
```

Comment:
i don't see the point in having this as an expression, since it is always the same value